### ISSUE TITLE
DPG should fix casing of parameters borrowed from OData

### ISSUE THREAD

**Issue by mikekistler** [Azure/autorest.typescript#1468]
The Azure REST API Guidelines recommend standard query parameters borrowed from OData. Some of these parameters have names that contain multiple words but are all lowercase run together, e.g. "orderby", "maxpagesize".

For client libraries, we want these names to be exposed as multi-word names using the appropriate language idiom, e.g. "orderBy" in .NET.

Our DPG generators (autorest-based or Cadl-based) should ensure the correct transformation of the parameter names in the REST API to idiomatic form for the target language.

**Comment by qiaozha** [comment]
I don't think this applies to RLC, because we don't have a mapper layer to map between the code name and the rest api name. @joheredi what do you think ?

**Comment by joheredi** [comment]
This doesn't apply to Rest Clients as you said, we just honor what the REST API call the parameters, in this case it would be the raw odata parameter names
